### PR TITLE
Make diff line-marker non-selectable

### LINKED ISSUE #4076: Copy/paste issues with diffs
Copying content from a diff like [this](https://try.gitea.io/go-gitea/gitea/commit/15f6ec96327d69d19fcce937811214e113341f58), I see inconsistent behaviour between browsers.

Firefox copies double newlines:
````
    claims := jwt.MapClaims{

      "repo": repo.ID,

      "op":   lfsVerb,

-     "exp":  now.Add(5 * time.Minute).Unix(),

+     "exp":  now.Add(setting.LFS.HTTPAuthExpiry).Unix(),

      "nbf":  now.Unix(),

    }

    if user != nil {
````

Safari copies line numbers:
````

claims := jwt.MapClaims{
--
269 | 269 | "repo": repo.ID,
270 | 270 | "op":   lfsVerb,
271 |   | -			"exp":  now.Add(5 * time.Minute).Unix(),
  | 271 | +			"exp":  now.Add(setting.LFS.HTTPAuthExpiry).Unix(),
272 | 272 | "nbf":  now.Unix(),
273 | 273 | }
274 | 274 | if user != nil {
````

Chrome seems alright:
````
    claims := jwt.MapClaims{
      "repo": repo.ID,
      "op":   lfsVerb,
-     "exp":  now.Add(5 * time.Minute).Unix(),
+     "exp":  now.Add(setting.LFS.HTTPAuthExpiry).Unix(),
      "nbf":  now.Unix(),
    }
    if user != nil {
````

I will probably attempt a fix later, but feel free to contribute as well.

### PR DESCRIPTION
**Pull request by zeripath**
This PR makes the first character of the diff line non-selectable. 

Fixes #4076

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by mrsdizzie** [comment]
While testing this I found out that while webkit obeys the "don't select" part, it still copies the contents if something around it has been selected:

Screen Shot 2019-06-22 at 11 04 00 PM

Then the copied text is:
```
# repo38
2	
+
3	
+Description for repo38
```

This appears to be an old issue unfortunately, https://bugs.webkit.org/show_bug.cgi?id=80159

I don't have this issue on Github or other similar sites, so I tried to see what was different:

It appears Github gets this to work through adding the line numbers and +/- using a pseudo class, like:
```
.blob-num:before {
content: attr(data-line-number);
}
```
Probably similar to what @silverwind was mentioning in the linked Issue.

I tried to create a really small example of how this might look to see if it would work on Safari and it does seem to:

https://codepen.io/mrsdizzie/pen/agWNwL

Knowledge of HTML/CSS is pretty limited though

**Comment by zeripath** [comment]
@mrsdizzie OK done.

File: public/css/index.css [hunk 1]
```diff
@@ -465,6 +465,7 @@ footer .ui.left,footer .ui.right{line-height:40px}
 .repository.file.list .non-diff-file-content .code-view table{width:100%}
 .repository.file.list .non-diff-file-content .code-view .lines-num{vertical-align:top;text-align:right;color:#999;background:#f5f5f5;width:1%;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}
 .repository.file.list .non-diff-file-content .code-view .lines-num span{line-height:20px;padding:0 10px;cursor:pointer;display:block}
+.repository.file.list .non-diff-file-content .code-view .lines-num[data-line-num]::before{content:attr(data-line-num);display:inline-block}
```
Comment (mrsdizzie) [inline public/css/index.css]: Should this have been in .diff-file-box not .non-diff-file-content ? Or maybe it belongs in both? I don't see the line numbers in a commit diff with the latest push

**Comment by mrsdizzie** [comment]
Aside from the comment above about line numbers the copy/paste now works in Safari, thanks!!

Also this is totally non blocking but since you are changing it:

I think it would nice if the +/- were in their own <td> as well, since it would/could give a bit more padding like in most the diffs here on Github (but not the comments on diffs strangely). Also on Safari when it is part of the same content it still lets you visually select the +/- chars on lines that also have selectable text, which might make people think that it will copy them as well.

It doesn't actually copy them any more so the real problem is solved and that is just a slight cosmetic issue at that point. Could be done in a separate PR if needed though.

File: public/css/index.css (hunk 1 above)
Comment (zeripath) [reply public/css/index.css]: Damn you're right. Don't know how I missed that.

**Comment by zeripath** [comment]
OK @mrsdizzie I think that's done

**Review by mrsdizzie — APPROVED** [review]
Nice, looks great!

**Comment by zeripath** [comment]
make lg-tm work

**Comment by silverwind** [comment]
Thanks, the `+/-` issue looks solved nicely. In Firefox, I do observe extraneous newlines being copied (4 newlines for each actual newline). Probably need to remove some whitespace inside the HTML tags holding the code lines.

**Comment by zeripath** [comment]
Hmm is that new? I haven't actually changed the template part that renders the code line.

**Comment by silverwind** [comment]
I think previously it was 3 extra newlines, with the introduction of `.lines-type-marker` it became 4. https://github.com/go-gitea/gitea/pull/7288 should solve it at least for Firefox.

**Pull request merged**